lib/manifest.ts

```diff
@@ -160,15 +160,15 @@ export class Manifest {
       return ver;
     }
 
-    const maxSupported = parseVersion(Manifest.version());
+    const maxSupported = semver.major(parseVersion(Manifest.version()));
     const actual = parseVersion(manifest.version);
 
-    // first validate the version should be accepted.
-    if (semver.gt(actual, maxSupported) && !options?.skipVersionCheck) {
+    // first validate the version should be accepted. all versions within the same minor version are fine
+    if (maxSupported !== semver.major(actual) && !options?.skipVersionCheck) {
```

Comment:
Did this allow lower major versions before? Now it will be confined to only the specific major version. 

Is that what we want, or do we still want to do a greater than check but just on the major version?

Reply:
follow up nit: if we are just limiting the version to a single major version, the "maximum version supported" language does not have much meaning any more

Reply:
I suppose we can allow lower but it doesn't actually make a ton of sense to do so. However, we only ever checked for higher before so I'll stick with that for now and we can revisit it later.